Initial commit

mhoppal opened the pull request.

mhoppal committed:
Initial commit

mhoppal committed:
Add configurable webhook address

mhoppal committed:
Update smoke tests with bigger timeout on webook

mhoppal committed:
Pass in webhook address to notification create

mhoppal committed:
asdf

mhoppal committed:
Update output

mhoppal committed:
Fix bind address

mhoppal committed:
Return when webhook is noticed

craigbr commented on smoke-test.go:
```diff
@@ -0,0 +1,212 @@
+		var err error
+		timeout, err = strconv.Atoi(timeoutSetting)
+		if err != nil {
+			timeout = 5
```
You are ignoring the configured value without telling the user

craigbr commented on smoke-test.go:
```diff
@@ -0,0 +1,212 @@
+
+var webhookTriggered bool = false
+var testsSucceeded int = 0
+var totalTests int = 6
```
I wonder if it makes more sense to increment totalTests inside every test. That way adding a new test doesn't require updating this value

mhoppal committed:
Add copyright to smoke tests

craigbr commented on smoke-test.go:
```diff
@@ -0,0 +1,212 @@
+			testsSucceeded++
+			return
+		}
+		time.Sleep(15 * time.Second)
```
I would wait less the 15 seconds. It is likely to happen within a second.

mhoppal committed:
Gofmt on file

mhoppal committed:
Add clean up method

Catches if previous runs failed in cleaning up its alarm
definition and notification method

mhoppal committed:
Change webhook wait time to 30 seconds

craigbr commented on smoke-test.go:
```diff
@@ -0,0 +1,279 @@
+}
+
+func cleanupPreviousRun() {
+	notificationName := "smoke_test_notification"
```
'smoke_test_notification' is defined twice. Should be a const

craigbr commented on smoke-test.go:
```diff
@@ -0,0 +1,279 @@
+	testsRun++
+
+	fmt.Println("TEST CLEANUP")
+	cleanup(alarmDefinitionID, notificationID)
```
Personally, I think you should only cleanup if everything else succeeded. Doing it always makes it harder to debug issues.

mhoppal replied:
done

craigbr commented on smoke-test.go:
```diff
@@ -0,0 +1,279 @@
+			}
+		}
+	}
+	alarmDefinitionName := "smoke_test_alarm"
```
Should be const

mhoppal replied:
moved to a global variable so each method can be use it. const cause issues trying to take address of when trying to pass in a pointer to it.

craigbr commented on smoke-test.go:
```diff
@@ -0,0 +1,279 @@
+		fmt.Printf("Error getting alarm definitions to delete potential left over alarm definition "+
+			"from previous runs - %s\n", err.Error())
+	} else {
+		alarmDefinitionID := ""
```
There should be 0 or 1 elements. Nothing to do if 0, if 1 delete it. Would be simpler

mhoppal committed:
Exit prior to cleanup if there was a failure
